packages/ember-data/lib/system/model/internal-model.js

```diff
@@ -663,6 +663,42 @@ InternalModel.prototype = {
     this._inFlightAttributes = Ember.create(null);
   },
 
+  /**
+    @method _changedKeys
+    @private
+  */
+  _changedKeys: function(updates) {
+    var changedKeys = [];
+
+    if (updates && typeof updates === 'object') {
+      var original, i, value, key;
+      var keys = Ember.keys(updates);
+      var length = keys.length;
+
+      original = merge({}, this._data);
+      original = merge(original, this._inFlightAttributes);
```

Comment:
`_attributes` do not need to be merged because we don't include it in the changed keys.